main/src/ca/uwaterloo/flix/language/phase/Stratifier.scala

```diff
@@ -776,4 +781,24 @@ object Stratifier extends Phase[Root, Root] {
     case (acc, _) => acc
   }
 
+
+  /**
+    * Returns the set of predicates that appears in the given row type `tpe`.
+    */
+  private def predicateSymbolsOf2(tpe: Type, acc: Map[Name.Pred, Int] = Map.empty): Option[Map[Name.Pred, Int]] = tpe match {
```

Comment:
Two options:

- Get the arity from the type.
- Get the arity directly from the syntax (e.g. the Predicate).

You are going for (1) which is more difficult, but might also be the better and more flexible solution.